scripts/odb_extractor_atoms.py

```diff
@@ -1,15 +1,187 @@
 # Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
 # For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
 
-import gzip
-import json
-import os
+#!/usr/bin/env python
+# coding: utf-8
 
+# Code for manipulating ODB Extractor json files and initial sequence atom definitions
+# Bryan Miller
+# 2021-11-24
+
+import os
+import sys
+import json
+import gzip
 import numpy as np
+import matplotlib.pyplot as plt
+from scipy.signal import find_peaks
+import requests
+
 from openpyxl import Workbook
 from openpyxl import load_workbook
 
-fpuinst = {'GNIRS': 'instrument:slitWidth', 'GMOS-N': 'instrument:fpu'}
+
+fpuinst = {'GSAOI': 'instrument:utilityWheel', 'GPI': 'instrument:observingMode', 'Flamingos2': 'instrument:fpu',
+           'NIFS': 'instrument:mask', 'GNIRS': 'instrument:slitWidth', 'GMOS-N': 'instrument:fpu',
+           'GMOS-S': 'instrument:fpu', 'NIRI': 'instrument:mask'}
+
+gpi_filter_wav = {'Y': 1.05, 'J': 1.25, 'H': 1.65, 'K1': 2.05, 'K2': 2.25}
+nifs_filter_wav = {'ZJ': 1.05, 'JH': 1.25, 'HK': 2.20}
+
+
+def find_filter(input, filter_dict):
+    """Match input string with filter list (in dictionary)"""
+
+    filter = ''
+    filters = list(filter_dict.keys())
+    for filt in filters:
+        if filt in input:
+            filter = filt
+            break
+    return filter
+
+
+def uniquelist(seq):
+    """Make a list of unique values"""
+    # http://stackoverflow.com/questions/480214/how-do-you-remove-duplicates-from-a-list-in-python-whilst-preserving-order
+    seen = set()
+    seen_add = seen.add
+    return [x for x in seq if not (x in seen or seen_add(x))]
```

Comment:
In Python 3.7+, this is faster:

```python
list(dict.fromkeys(seq))
```

No more need to use sets or for comprehensions.